### PR TITLE
add comm command #346

### LINKED ISSUE #346: add a ref page to `comm`
`comm` is a part of the family of comparison tools but a reference page can help clarify the differences between `diff`, `cmp`, `comm`.

suggested by ppannuto in #343

### PR DESCRIPTION
**Pull request by JunhaoWang**
# Advanced HW 5 submission

Closes #346 

Name: Junhao Wang
Uniqname: junhao

### CODE REVIEW

File: _commands/basics/comm.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+---
+
+comm
+---
+Compares two sorted files line by line and writes three columns to standard output.
+<!-- one line explanation would go here -->
+
+<!-- minimal example -->
+~~~ bash
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: This example is too long to be displayed on the reference page. Can a shorter example be given here?

File: _commands/basics/comm.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+comm -i words1.txt words2.txt
+            Apple
+            Banana
+            Orange
+~~~
+
+<!--more-->
+
+### Useful Options / Examples
+
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: More examples need to be given to show some use cases of the `comm` command.

File: _commands/basics/comm.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+comm
+---
+Compares two sorted files line by line and writes three columns to standard output.
+<!-- one line explanation would go here -->
+
+<!-- minimal example -->
+~~~ bash
+cat words1.txt
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: What do the three columns mean? This isn't explained anywhere in the information given.

File: _commands/basics/comm.md (hunk 1 above)
Comment (JunhaoWang) [reply _commands/basics/comm.md]: Modified the example to be smaller

File: _commands/basics/comm.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+comm
+---
+Compares two sorted files line by line and writes three columns to standard output.
+<!-- one line explanation would go here -->
+
+<!-- minimal example -->
+~~~ bash
+# content of file1.txt
+$ cat file1.txt
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: Show the contents of file1.txt and file2.txt next to each other in the same lines. That way they don't take up as much space in the example.

File: _commands/basics/comm.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+abc
+def
+ghi
+
+# content of file2.txt
+$ cat file2.txt
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: Add this output next to the output above

File: _commands/basics/comm.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+abc
+ghi
+klm
+
+# compare file1.txt and file2.txt: first column showing unique lines of first file,
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: This explanation can go after the "`<!--more-->`" comment, to shorten the amount of information shown on the reference page.

File: _commands/basics/comm.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+ghi
+
+# content of file2.txt
+$ cat file2.txt
+abc
+ghi
+klm
+
+# compare file1.txt and file2.txt: first column showing unique lines of first file, 
+# second column showing unique lines of second file, thrird column showing matched
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: spelling error: "thrird" -> "third"

File: _commands/basics/comm.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	abc
+def
+	ghi
+
+# suppress third column
+$ comm -3 file1.txt file2.txt
+def
+	klm
+~~~
+### Useful Options / Examples
```
Comment (cameron-gagnon) [inline _commands/basics/comm.md]: Add these headers to the examples given above. That way it separates out each example better and is easier to follow.

**Pull request closed without merging**